gflownet/envs/grid.py

```diff
@@ -58,9 +59,12 @@ def __init__(
         elif self.proxy_state_format == "oracle":
             self.statebatch2proxy = self.statebatch2oracle
             self.statetorch2proxy = self.statetorch2oracle
-        # Set up proxy
-        self.proxy.n_dim = self.n_dim
-        self.proxy.setup()
+        # Set up oracle
+        self.oracle.n_dim = self.n_dim
+        self.oracle.setup()
+        # TODO: change to something more general. this would be invalid if num_fid>5
+        self.invalid_action = self.n_dim + 5
+        self.proxy_factor = 1.0
```

Comment:
need to make this dependant on proxy not oracle

Reply:
Your comment is to explain your changes? If so, I agree.